src/spec.c

```diff
@@ -2860,6 +2912,22 @@ int IndexSpec_RegisterType(RedisModuleCtx *ctx) {
   return REDISMODULE_OK;
 }
 
+static inline FieldExpiration* getHashFieldExpirationTime(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString *key) {
+  arrayof(RedisModuleString *) fields = array_newlen(RedisModuleString *, spec->numFields);
+  for (t_fieldIndex field = 0; field < spec->numFields; ++field) {
+    FieldSpec *f = spec->fields + field;
+    RS_LOG_ASSERT(f->index < spec->numFields, "index is outside allowed range");
```

Comment:
Relevant here? The index should be set on field creation and validated there. I think we can trust the value here